cookbooks/aws-parallelcluster-environment/recipes/init/config_default_user_home.rb

```diff
@@ -63,19 +63,21 @@
 end
 
 # Data integrity check and cleanup for temporary backup and original home directory
-# 1. Verifies data integrity by comparing the temporary backup directory and the new local home directory.
+# 1. Verifies data integrity by comparing the original home directory and the new local home directory.
 # 2. If the data integrity check passes, it removes both the temporary backup directory and the original home directory.
 # 3. If the data integrity check fails, it outputs an error message and exits with an error code 1.
 bash "Verify data integrity for #{node['cluster']['cluster_user_home']}" do
   user 'root'
   group 'root'
   code <<-EOH
     diff_output=$(diff -r #{node['cluster']['cluster_user_home']} #{node['cluster']['cluster_user_local_home']})
-    if [ $? -eq 0 ]; then
+    if [[ $diff_output != *"Only in #{node['cluster']['cluster_user_home']}"* ]]; then
```

Comment:
*Best Practice (not a blocker, but to keep in mind)*
We should always refer positive check over negative checks because it improves readability and, as a consequence, maintainability.

It's easier to understand a positive check than a negative one.